packages/server/utils/analytics/amplitude/AmplitudeAnalytics.ts

```diff
@@ -0,0 +1,29 @@
+import {init, track} from '@amplitude/analytics-node'
+import {AnalyticsEvent} from '../analytics'
+import PROD from '../../../PROD'
+import getDataLoader from '../../../graphql/getDataLoader'
+
+const {AMPLITUDE_WRITE_KEY} = process.env
+
+/**
+ * Wrapper for amplitude providing a more typesafe interface
+ */
+export class AmplitudeAnalytics {
+  constructor() {
+    init(AMPLITUDE_WRITE_KEY || 'x', {
+      flushQueueSize: PROD ? 20 : 1,
+      optOut: !AMPLITUDE_WRITE_KEY
+    })
+  }
+
+  async track(userId: string, event: AnalyticsEvent, properties?: Record<string, any>) {
+    // used as a failsafe for PPMIs
+    if (!AMPLITUDE_WRITE_KEY) return
+    const user = await getDataLoader().get('users').load(userId)
```

Comment:
@tianrunhe I've updated this and the relevant Segment code to use the dataloader, but I'm not sure I've done it correctly.  Everywhere else seems to call the dataloader through the GraphQL execution environment, whereas this is bypassing that.  Does this work? Or do we need to run all of our tracking through the GraphQL executors?

Reply:
Unfortunately no... `await getDataLoader()` creates a new DataLoaderCache instance each time it's called so essentially it still hits PG every time. Porting the dataLoader from GraphQL seems overkill.
I saw in `segmentIo.ts` today we are doing `await getUserById(userId)` anyway so maybe I was over concerned. Let's just keep it as-is then!